### PR TITLE
fix(client): get sound value from redux to play sound

### LINKED ISSUE #46372: Campfire Mode doesn't trigger on multiple browsers unless you turn it on and off
### Describe the Issue

Campfire Mode was not playing on my account. I did find a fix, but still wanted to report! Please note it could be affecting more than one page, but I only tested on the new responsive web design! 

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/

### Steps to Reproduce

1. I'm logged in on my computer 
2. Under settings, Campfire Mode is enabled. 
3. My volume is turned up!
4. I go to the (New) Responsive Web Design course
5. The serenade of the Guitar does not play as I type or click on items :(

### Expected behavior

1. I'm logged in on my computer 
2. Under settings, Campfire Mode is enabled. 
3. My volume is turned up!
4. I go to the (New) Responsive Web Design course
5. The serenade of the Guitar plays!

### Screenshots

_No response_

### System

- Mac OS Big Sur 11.6.5 running Safari
- Windows 10 running Google Chrome 

### Additional context

I found a fix while poking around! 

1. I go to my settings and turn off and on Campfire mode.
2. The serenade of the Guitar DOES play as I type and click on items :)

### PR DESCRIPTION
**Pull request by nayabatir1**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #46372 
Closes #48106

This 2nd method of solving issue.
Here I've removed `localstorage` dependency from `playTone`. Sound value is grabbed directly from redux and audio is played accordingly.

### CODE REVIEW

**Comment by ieahleen** [comment]
Please take a moment to give a name to the PR that follows the guidelines: https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title

**Review by raisedadead — CHANGES_REQUESTED** [review]
Why are you re-arranging the unrelated imports? 

Other than that, I like the idea of not needing to use the localStorage, but others are better qualified to confirm.

File: client/src/utils/tone/index.ts
Comment (ojeytonwilliams) [inline client/src/utils/tone/index.ts:2]:
```suggestion
```
What's the rationale for this change?  We deliberately do not statically import `tone` because it's a large library that not all users need to view the site.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Comment by nayabatir1** [comment]
> Please take a moment to give a name to the PR that follows the guidelines: https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title

I've updated pr title

File: client/src/utils/tone/index.ts
Comment (nayabatir1) [reply client/src/utils/tone/index.ts:2]: I used this way of import b'cuz it is mentioned in npm.
I did some research and found a better way to  import  is `import { loaded, Player } from 'tone';`.  
Taking advantage of [tree shaking](https://webpack.js.org/guides/tree-shaking/).
If it seem fit I can change it or revert previous way importing.

**Comment by raisedadead** [comment]
You do not need to work on the other PR (the other solution). Just wait on some conclusive comments from others. 

This is likely the version we want. You are welcome to hang on to the other if you want.

**Comment by nayabatir1** [comment]
> You do not need to work on the other PR (the other solution). Just wait on some conclusive comments from others.
> 
> This is likely the version we want. You are welcome to hang on to the other if you prefer - but if there is anything the maintainers like, that is less noise.

Sorry if it bothers you.
I found a better solution so I thought why not implement it.

**Comment by ojeytonwilliams** [comment]
https://github.com/freeCodeCamp/freeCodeCamp/pull/48106#pullrequestreview-1146486113 applies here, too. I hadn't realised there were two PRs solving the same issue.

> I found a better solution so I thought why not implement it.

The reason is that small, focused PRs are easier to understand. This makes it easier for reviewers to see if there are any issues that need to be addressed before merging. Also, if we later find that there are problems, it's easier to debug them.

**Comment by raisedadead** [comment]
> Sorry if it bothers you.

That's not the problem. 

The issue is you have two PRs, making it easier for multiple people to review your work harder. You may be working on a couple of issues; the maintainers deal with a dozen PRs. It is easy for us to lose context.

We get that you have two different approaches, and the simplest thing to do in such cases would be to keep the discussion on the most favored approach (this PR) and not work on the other one until you are asked to.

It would have been even better to get a confirmation on the main issue - but we are past that before you started.

Let's focus on working on the review of this PR.

**Comment by naomi-lgbt** [comment]
> I did some research and found that audio can be played without any dependency. I think we should definitely use this method

The purpose of using Tone specifically was the ability to **easily** generate dynamic audio in the editor - while it would be possible to do so with the raw Audio context, IMHO it's not worth the overhead.

**Comment by nayabatir1** [comment]
@ojeytonwilliams can I create checkout of this branch for removing tone package. Checkout of main branch will result in merge conflict as both branch modify same function `playTone`

**Comment by ojeytonwilliams** [comment]
Sorry, @nayabatir1, I hadn't realised why we were using Tone specifically.  @naomi-lgbt understands this issue better than I do, so I will defer to her.

For this PR, please could you keep the changes to those needed to fix the original issue, https://github.com/freeCodeCamp/freeCodeCamp/issues/46372?  There's no need to re-arrange the imports or change to statically importing to address that.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Because the logic to check `store` has been removed from `playTone`, there are a few different areas where we'll need to check the user's settings:

- client/src/components/Donation/donation-modal.tsx
- client/src/components/Flash/redux/index.ts
- client/src/components/settings/sound.tsx
- client/src/templates/Challenges/redux/execute-challenge-saga.js

**Comment by nayabatir1** [comment]
@naomi-lgbt I looked into all the mentioned files.
All functions must be converted to React components in order to access redux store values.
For simplicity, I think we should consider this [pr](https://github.com/freeCodeCamp/freeCodeCamp/pull/48106)

**Comment by naomi-lgbt** [comment]
We can certainly consider that approach, though we'd need to not remove Tone in the process.

**Comment by nayabatir1** [comment]
> We can certainly consider that approach, though we'd need to not remove Tone in the process.

@naomi-lgbt I've updated [pr](https://github.com/freeCodeCamp/freeCodeCamp/pull/48106).

**Pull request closed without merging**
